jaeger/cmd/check.go

```diff
@@ -0,0 +1,232 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/linkerd/linkerd2/pkg/healthcheck"
+	"github.com/linkerd/linkerd2/pkg/k8s"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+
+	// jaegerExtensionName is the name fo jaeger extension
+	jaegerExtensionName = "linkerd-jaeger"
+
+	// linkerdJaegerExtensionCheck adds checks related to the jaeger extension
+	linkerdJaegerExtensionCheck healthcheck.CategoryID = jaegerExtensionName
+)
+
+var (
+	jaegerNamespace string
+)
+
+type checkOptions struct {
+	wait   time.Duration
+	output string
+}
+
+func jaegerCategory() (*healthcheck.Category, error) {
+
+	kubeAPI, err := k8s.NewAPI(kubeconfigPath, kubeContext, impersonate, impersonateGroup, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	checkers := []healthcheck.Checker{}
+	checkers = append(checkers,
+		*healthcheck.NewChecker("collector service account exists", "", false, true, time.Time{}, false).
```

Comment:
Also I kinda like better how we have things set up in `pkg/healthcheck/healthcheck.go` where you just instantiate the `Checker` struct. Using a long function call like this forces one to go dig into the declaration to see what each argument means.

Reply:
Shouldn't all these checks be `fatal`?

Reply:
> how we have things set up in pkg/healthcheck/healthcheck.go where you just instantiate the Checker struct. 

This is not possible here because those fields are not public, so not accessible here. Hence this helper function here. 🤔 To make them public would be a huge change. WDYT?

Reply:
> Shouldn't all these checks be fatal?

Currently, They are not and hence I did  the same. I think this was done so that you can get full view on whats happening in the extension. Should they fail instead? 🤔

Reply:
> > how we have things set up in pkg/healthcheck/healthcheck.go where you just instantiate the Checker struct.
> 
> This is not possible here because those fields are not public, so not accessible here. Hence this helper function here.  To make them public would be a huge change. WDYT?

I'm ok making those fields public now that they're used in other packages. The diff certainly will get bigger, but I think it's manageable given it's just string changes.

Reply:
Hmm, The Change is manageable, but it could break the API that these structs expose right? to be able to access them anywhere and that creates further expectations on future changes. I remember, there was some discussions around keeping private fields private wherever possible. 

@adleong WDYT?